### PR TITLE
Add support for Oracle Cloud in deploy

### PR DESCRIPTION
**Pull request by bruno-garbaccio**
Following [issue #6849,](https://github.com/triton-inference-server/server/issues/6849), here is a proposal to have Triton server working on the OKE service of Oracle Cloud.

The directory deploy/oci is based on deploy/aws with some minor adjustments to have it run on Oracle Cloud.

### CODE REVIEW

**Comment by rmccorm4** [comment]
Hi @bruno-garbaccio, thanks for submitting this PR. Have you submitted a CLA as described here: https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla?

**Comment by rmccorm4** [comment]
Thanks! CC @nealvaidya @mengdong @nv-kmcgill53 @whoisj could you give this a look over when you get a chance?

**Comment by bruno-garbaccio** [comment]
Hi @rmccorm4, @nealvaidya , @mengdong , @nv-kmcgill53 @whoisj  I know you guys are busy with a ton of things but do you think this could be reviewed this week? I would like to have a communication in one of our Oracle Blogs about the support of OCI for the deployment and we have a deadline for this if we want it to happen during GTC. I run the tests in my OCI tenant,took the "aws" repo as a basis and try to respect the README.md format that was set for "aws". Do not hesitate to reach out for questions!

**Comment by mengdong** [comment]
the helm chart and instruction looks good to me!

**Comment by bruno-garbaccio** [comment]
Hi, I saw the pre-commit run failed at line 160 in the readme, at the level of the url:  s3://https://<OCI_NAMESPACE>.compat.objectstorage.<OCI_REGION>.oraclecloud.com:443/triton-inference-server-repository

This url comes from the documentation https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_repository.html and in the OCI case there are some placeholders to fill. I don't have a full error log. Can you tell me if there's anything I could do?

File: deploy/oci/README.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+bash /var/run/oke-init.sh
+sudo /usr/libexec/oci-growfs -y
+```
+
+
+## Installing Helm
+
+### Using Cloud Shell from OCI Web Console
+
+It is possible to access your OKE Cluster [directly from the OCI Web Console](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengaccessingclusterkubectl.htm).
```
Comment (rmccorm4) [inline deploy/oci/README.md:72]: Pre-commit is complaining about trailing whitespace at the end of the line
```suggestion
It is possible to access your OKE Cluster [directly from the OCI Web Console](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengaccessingclusterkubectl.htm).
```

File: deploy/oci/README.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+echo -n 'REGION' | base64
+```
+```
+echo -n 'SECRECT_KEY_ID' | base64
+```
+```
+echo -n 'SECRET_ACCESS_KEY' | base64
+```
+
+You also need to adapt _modelRepositoryPath_ in values.yaml to your [namespace](https://docs.oracle.com/en-us/iaas/Content/Object/Tasks/understandingnamespaces.htm) and [OCI region](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm)
```
Comment (rmccorm4) [inline deploy/oci/README.md:141]: Pre-commit is complaining about trailing whitespace at the end of the line

```suggestion
You also need to adapt _modelRepositoryPath_ in values.yaml to your [namespace](https://docs.oracle.com/en-us/iaas/Content/Object/Tasks/understandingnamespaces.htm) and [OCI region](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm).
```

**Comment by rmccorm4** [comment]
Hi @bruno-garbaccio, I believe pre-commit is just failing for trailing whitespace - I added some suggestions above to fix it. Otherwise, LGTM and I can approve/merge.

**Review by rmccorm4 — APPROVED** [review]

**Comment by rmccorm4** [comment]
Dong, J, and Neal all gave their seal of approval, so merging.

**Pull request merged**